component/vdr/orb/README.md

```diff
@@ -0,0 +1,245 @@
+}
+
+func (k *keyRetrieverImpl) GetNextRecoveryPublicKey(didID string) (crypto.PublicKey, error) {
+	return k.nextRecoveryPublicKey, nil
+}
+
+func (k *keyRetrieverImpl) GetNextUpdatePublicKey(didID string) (crypto.PublicKey, error) {
+	return k.nextUpdatePublicKey, nil
+}
+
+func (k *keyRetrieverImpl) GetSigningKey(didID string, ot orb.OperationType) (crypto.PrivateKey, error) {
+	if ot == orb.Update {
+		return k.updateKey, nil
+	}
+
+	return k.recoverKey, nil
+}
+
+
+keyRetrieverImpl := &keyRetrieverImpl{}
+
+vdr, err := orb.New(keyRetrieverImpl, orb.WithDomain("https://testnet.devel.trustbloc.dev"))
+	if err != nil {
+		return err
+}
+```
+
+## Create DID
+For creating DID use vdr create and pass DID document. To discover orb instance there are two ways explicitly or
+through domain.
+
+```
+import (
+"crypto"
+"crypto/ed25519"
+"crypto/rand"
+"fmt"
+
+ariesdid "github.com/hyperledger/aries-framework-go/pkg/doc/did"
+"github.com/hyperledger/aries-framework-go/pkg/doc/jose"
+vdrapi "github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr"
+
+"github.com/hyperledger/aries-framework-go-ext/component/vdr/orb"
+)
+
+recoveryKey, recoveryKeyPrivateKey, err := ed25519.GenerateKey(rand.Reader)
+if err != nil {
+	return err
+}
+
+updateKey, updateKeyPrivateKey, err := ed25519.GenerateKey(rand.Reader)
+if err != nil {
+	return err
+}
+
+didPublicKey, _, err := ed25519.GenerateKey(rand.Reader)
+if err != nil {
+	return err
+}
+
+jwk, err := jose.JWKFromKey(didPublicKey)
+if err != nil {
+	return err
+}
+
+vm,err:=ariesdid.NewVerificationMethodFromJWK("key1", "Ed25519VerificationKey2018", "", jwk)
+if err != nil {
+	return err
+}
+
+didDoc := &ariesdid.Doc{}
+
+// add did keys
+didDoc.Authentication = append(didDoc.Authentication, *ariesdid.NewReferencedVerification(vm,
+		ariesdid.Authentication))
+
+// add did services
+didDoc.Service = []ariesdid.Service{{ID: "svc1", Type: "type", ServiceEndpoint: "http://www.example.com/"}}
+
+// create did
+createdDocResolution, err := vdr.Create(didDoc,
+		vdrapi.WithOption(orb.RecoveryPublicKeyOpt, recoveryKey),
+		vdrapi.WithOption(orb.UpdatePublicKeyOpt, updateKey),
+		// No need to use this option because we already use domain
+		// vdrapi.WithOption(orb.OperationEndpointsOpt, []string{"https://orb-1.devel.trustbloc.dev/sidetree/v1/operations"}),
+		vdrapi.WithOption(orb.AnchorOriginOpt, "https://orb-2.devel.trustbloc.dev/services/orb"))
+if err != nil {
+	return err
+}
+
+fmt.Println(createdDocResolution.DIDDocument.ID)
+
+// recovery private key be will used to sign next recovery request
+keyRetrieverImpl.recoverKey = recoveryKeyPrivateKey
+// update private key will be used to sign next update request
+keyRetrieverImpl.updateKey = updateKeyPrivateKey
+
+
+// if you need did to be discoverable just add domain to did URI (did:orb:123 => did:orb:domain.com:123)
+discoverableDID := strings.ReplaceAll(createdDocResolution.DIDDocument.ID, "did:orb", "testnet.devel.trustbloc.dev")
```

Comment:
Should it be did:orb:testnet.devel.trustbloc.dev?

Reply:
yes